+++ ruletypes.py	2018-01-11 14:38:02.866838792 +0100
@@ -2,6 +2,7 @@
 import copy
 import datetime
 import sys
+import array
 
 from blist import sortedlist
 from util import add_raw_postfix
@@ -153,6 +154,38 @@
             return True
         return False
 
+class ConsecutiveGrowthRule(CompareRule):
+    required_options = frozenset(['compare_key'])
+    arr = array.array('d')
+
+    def set_off_alarm(self):
+	if self.arr.buffer_info()[1] < 3:
+		return False
+	if (self.arr[-1] > self.arr[-2]) and (self.arr[-2] > self.arr[-3]):
+		return True
+	return False
+
+    def add_value(self, value, max_size):
+	try:
+        	self.arr.append(value)
+	except TypeError:
+		elastalert_logger.info("Ignoring value since it is not a number")
+		return False
+        while self.arr.buffer_info()[1] > max_size:
+                self.arr.pop(0)
+	if self.arr.buffer_info()[1] == max_size:
+		return True
+	return False
+
+    def compare(self, event):
+	changed = False
+	value = lookup_es_key(event, self.rules['compare_key'])
+	if self.add_value(value, 3):
+		changed = self.set_off_alarm()
+
+	elastalert_logger.info("%s" % (str(self.arr.tolist())))
+	return changed
+
 
 class ChangeRule(CompareRule):
     """ A rule that will store values for a certain term and match if those values change """
